### ISSUE TITLE
OpenSSL creates PKCS#12 files with duplicate certificates

### ISSUE THREAD

**Issue by tomato42** [openssl/openssl#15983]
Using current master, 92d7120c60cea7c9

When I use openssl to create a PKCS#12 file with just a certificate and a friendly name, it includes the certificate in the file twice.
This is a regression from 1.1.1k and impacts interoperability with applications that use NSS, as the first certificate in the PKCS#12 files doesn't have a "friendly name", so it doesn't get imported to the NSS database with one.

Reproducer:
```
openssl req -x509 -newkey rsa -keyout /tmp/localhost.key -out /tmp/localhost.crt -subj /CN=localhost -nodes -batch
openssl pkcs12 -export -out /tmp/localhost.p12 -in /tmp/localhost.crt -caname localhost -nokeys -passout pass: -certpbe NONE
openssl pkcs12 -info -in /tmp/localhost.p12 -passin pass:
openssl asn1parse -inform der -in /tmp/localhost.p12 -strparse 26 -strparse 23
```

the `pkcs12 -info` lists two certificates, first one with no bag attributes, `asn1parse` lists two certBag objects

**Comment by t8m** [comment]
Any idea if this is a recent regression?

**Comment by tomato42** [comment]
it's in 3.0.0 since at least alpha16, it's not there in 1.1.1k, haven't checked any other versions

**Comment by mattcaswell** [comment]
Offending commit seems to be 1d6c86709c72

**Comment by mattcaswell** [comment]
Fix for this in #16001
